test/functional/kernel/region/tests/test-kernel-region.hpp

```diff
@@ -0,0 +1,95 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef __TEST_KERNEL_REGION_HPP__
+#define __TEST_KERNEL_REGION_HPP__
+
+#include "RAJA/RAJA.hpp"
+
+#include "camp/resource.hpp"
+
+#include "gtest/gtest.h"
+
+TYPED_TEST_SUITE_P(KernelRegionTest);
+template <typename T>
+class KernelRegionTest : public ::testing::Test
+{
+};
+
+template <typename INDEX_TYPE, typename WORKING_RES, typename EXEC_POLICY>
+void KernelBasicRegionTest(INDEX_TYPE first, INDEX_TYPE last)
+{
+  camp::resources::Resource work_res{WORKING_RES()};
+
+  //
+  // Set some local variables and create some segments for using in tests
+  //
+  const INDEX_TYPE N = last - first;
+  
+  RAJA::TypedRangeSegment<INDEX_TYPE> rseg(first, last);
+
+  INDEX_TYPE* work_array1 = work_res.allocate<INDEX_TYPE>(N);
+  INDEX_TYPE* work_array2 = work_res.allocate<INDEX_TYPE>(N);
+  INDEX_TYPE* work_array3 = work_res.allocate<INDEX_TYPE>(N);
+
+  work_res.memset( work_array1, 0, sizeof(INDEX_TYPE) * N );
+  work_res.memset( work_array2, 0, sizeof(INDEX_TYPE) * N );
+  work_res.memset( work_array3, 0, sizeof(INDEX_TYPE) * N );
+
+  camp::resources::Resource host_res{camp::resources::Host()};
+
+  INDEX_TYPE* check_array = host_res.allocate<INDEX_TYPE>(N);
+
+  host_res.memset( check_array, 0, sizeof(INDEX_TYPE) * N );
+
+  RAJA::kernel<EXEC_POLICY>(
+
+    RAJA::make_tuple(rseg),
+
+    [=] (INDEX_TYPE i) {
+      work_array1[i - first] = 50;
+    },
+
+    [=] (INDEX_TYPE i) {
+      work_array2[i - first] = 100;
+    },
+
+    [=] (INDEX_TYPE i) {
+      work_array3[i - first] = work_array1[i - first] + 
+                               work_array2[i - first] + 1;
```

Comment:
If this is passing with the current no-wait policy (I think we are missing a sync threads above), I suspect that iterations across the different loops are being partitioned and distributed in the same manner, so there is no risk of one thread touching data from a different thread. It could be useful to traverse work_array2 backwards to make sure threads are synchronizing when needed.

Reply:
@artv3 good point.  As I've written these tests, there should be no need for a sync threads call. However, I will add a couple more tests that require it to test that also. Thanks.